limbo/testcases/rfc5280.py

```diff
@@ -611,3 +612,425 @@ def leaf_ku_keycertsign(builder: Builder) -> None:
 
     builder = builder.client_validation()
     builder.trusted_certs().peer_certificate(leaf).fails()
+
+
+@testcase
+def ca_nameconstraints_permitted_dns_mismatch(builder: Builder) -> None:
+    """
+    Produces the following **invalid** chain:
+
+    ```
+    root -> leaf
+    ```
+
+    The root contains a NameConstraints extension with a permitted dNSName
+    "example.com", whereas the leaf certificate has a SubjectAlternativeName with a
+    dNSName of "not-example.com".
+    """
+    root = builder.root_ca(
+        name_constraints=ext(
+            x509.NameConstraints([x509.DNSName("example.com")], None), critical=True
```

Comment:
Pedantic: let's use explicit kwargs for `NameConstraints` to make the behavior here clearer:

```suggestion
            x509.NameConstraints(permitted_subtrees=[x509.DNSName("example.com")], None), critical=True
```

(and same for all other testcases.)

Reply:
Same for `excluded_subtrees`, of course 🙂